source/code/source/playfab/PlayFabIOSHttpPlugin.mm

```diff
@@ -91,38 +91,42 @@ void Cancel()
 
     void PlayFabIOSHttpPlugin::MakePostRequest(std::unique_ptr<CallRequestContainerBase> requestContainer)
     {
-        std::shared_ptr<RequestTask> requestTask = nullptr;
-        try
-        {
-            requestTask = std::make_shared<RequestTask>();
-            requestTask->Initialize(requestContainer);
-        }
-        catch (const std::exception& ex)
+        CallRequestContainer* container = dynamic_cast<CallRequestContainer*>(requestContainer.get());
+        if (container != nullptr && container->ValidateSettings())
         {
-            PlayFabPluginManager::GetInstance().HandleException(ex);
-        }
-        catch (...)
-        {
-
-        }
-        if(requestTask != nullptr)
-        { // LOCK httpRequestMutex
-            std::unique_lock<std::mutex> lock(httpRequestMutex);
-            requestTask->state = RequestTask::State::Pending;
-            pendingRequests.push_back(std::move(requestTask));
-            if(workerThread == nullptr)
+            std::shared_ptr<RequestTask> requestTask = nullptr;
+            try
             {
-                threadRunning = true;
-                workerThread = std::make_unique<std::thread>(&PlayFabIOSHttpPlugin::WorkerThread, this);
+                requestTask = std::make_shared<RequestTask>();
+                requestTask->Initialize(requestContainer);
             }
-        } // UNLOCK httpRequestMutex
+            catch (const std::exception& ex)
+            {
+                PlayFabPluginManager::GetInstance().HandleException(ex);
+            }
+            catch (...)
+            {
+
+            }
```

Comment:
add line before conditional statements

Reply:
Generally, I prefer not to change lines that aren't part of this PR.
I'll do this one since it's adjacent to relevant changes and fast, but I'd resist doing lots of them.

---
In reply to: [343404249](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r343404249) [](ancestors = 343404249)